rayon-core/src/thread_pool/mod.rs

```diff
@@ -161,8 +212,43 @@ impl ThreadPool {
         unsafe { spawn::spawn_in(op, &self.registry) }
     }
 
-    /// Spawns an asynchronous future in this thread-pool. See
-    /// `spawn_future()` for more details.
+    /// Spawns an asynchronous future in the thread pool. `spawn_future()` will inject 
+    /// jobs into the threadpool that are not tied to your current stack frame. This means 
+    /// `ThreadPool`'s `spawn` methods are not scoped. As a result, it cannot access data
+    /// owned by the stack.
+    ///
+    /// `spawn_future()` returns a `RayonFuture<F::Item, F::Error>`, allowing you to chain
+    /// multiple jobs togther.
+    ///
+    /// ## Using `spawn_future()`
+    ///
+    /// ```rust
+    ///    # extern crate rayon_core as rayon;
+    ///    extern crate futures;
+    ///    
+    ///    use futures::{future, Future};
+    ///    use std::sync::{Arc, Mutex};
+    ///    # fn main() {
+    ///    let data = Arc::new(Mutex::new(String::from("Hello,")));
+    ///    let pool = rayon::ThreadPool::new(rayon::Configuration::new().num_threads(8)).unwrap();
+    ///
+    ///    let a = pool.spawn_future(future::lazy({
+    ///                                               let data = data.clone();
+    ///                                               move || Ok::<_, ()>(data)
+    ///                                           }));
+    ///    let b = pool.spawn_future(a.map(|data| {
```

Comment:
I took this example from one of the test cases for `spawn_future`. (https://github.com/nikomatsakis/rayon/blob/master/rayon-core/src/spawn/test.rs#L53) I should have given more thought in hindsight though. Thank you for spending the time on this.